proposed/2021/DotNetAudit.md

```diff
@@ -0,0 +1,296 @@
+
+There are a number of auditing tools in other ecosystems that are third-party / developed by the community. For other ecosystems, these features are built into the first-party CLI experiences.
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+There already exists a dotnet CLI commands called `dotnet list package` with options such as `--vulnerable, --deprecated, --outdated` which will list any known vulnerabilities, deprecations, and outdated packages in a project or solution. These options currently cannot be combined. Although this provides valuable information to understand the state of your dependencies, there does not exist a tool that allows you to quickly audit a project/solution & provide a way to act further on.
+
+When a developer is prompted with information such as a known vulnerability, deprecated package, or outdated version, they should have a clear understanding of how to best proceed with the information provided. In most cases, simply updating the dependency will suffice.
+
+Additionally, there already exists many third-party solutions that try to solve this problem with varying degrees of success. Some solutions can help alert about known vulnerabilities and some can help alert about outdated packages. No single tool seems to check for a "healthy" dependency list & that's another reason why we're looking to combine these experiences into a single command.
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevent to this proposal? -->
+- [snyk](https://snyk.io/)
+- [npm audit](https://docs.npmjs.com/cli/v7/commands/npm-audit)
+- [cargo audit](https://github.com/RustSec/cargo-audit)
+- [dotnet outdated](https://github.com/dotnet-outdated/dotnet-outdated)
+- [dotnet retire](https://github.com/retirenet/dotnet-retire)
+- [NuGet Defense](https://github.com/digitalcoyote/NuGetDefense)
+  
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+- Should the command be named `audit` or `check`?
+  - `audit` is the more consistent name for package manager tooling with other ecosystems.
+  - `check` is the more consistent name with dotnet CLI.
+- Should this command only audit `vulnerabilities`? Or should it audit & be proactive as to `vulnerabilities`, `deprecations`, and `outdated` packages?
```

Comment:
I think vulnerabilities and deprecations should be accounted for. 

Outdated packages is the one that gives me a pause (at least by default), due to the fact updating packages to the latest is not necessarily always the right solution.

Reply:
Makes sense. I'd like to see what other thoughts are about this 3-in-1 tool and then can iterate further.